core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -297,46 +297,37 @@ public ListFeatureSetsResponse listFeatureSets(ListFeatureSetsRequest.Filter fil
    *     filter
    */
   public ListFeaturesResponse listFeatures(ListFeaturesRequest.Filter filter) {
-    try {
-      String project = filter.getProject();
-      List<String> entities = filter.getEntitiesList();
-      Map<String, String> labels = filter.getLabelsMap();
+    String project = filter.getProject();
+    List<String> entities = filter.getEntitiesList();
+    Map<String, String> labels = filter.getLabelsMap();
 
-      checkValidCharactersAllowAsterisk(project, "project");
+    checkValidCharactersAllowAsterisk(project, "project");
 
-      // Autofill default project if project not specified
-      if (project.isEmpty()) {
-        project = Project.DEFAULT_NAME;
-      }
+    // Autofill default project if project not specified
+    if (project.isEmpty()) {
+      project = Project.DEFAULT_NAME;
+    }
 
-      // Currently defaults to all FeatureSets
-      List<FeatureSet> featureSets =
-          featureSetRepository.findAllByNameLikeAndProject_NameOrderByNameAsc("%", project);
-      // TODO: List features in Feature Tables.
+    // Currently defaults to all FeatureTables
+    List<FeatureTable> featureTables = tableRepository.findAllByProject_Name(project);
 
-      ListFeaturesResponse.Builder response = ListFeaturesResponse.newBuilder();
-      if (entities.size() > 0) {
-        featureSets =
-            featureSets.stream()
-                .filter(featureSet -> featureSet.hasAllEntities(entities))
-                .collect(Collectors.toList());
-      }
+    ListFeaturesResponse.Builder response = ListFeaturesResponse.newBuilder();
+    if (entities.size() > 0) {
+      featureTables =
+          featureTables.stream()
+              .filter(featureTable -> featureTable.hasAllEntities(entities))
```

Comment:
We should check that entities are exactly match to whatever specified in filter, not a subset